presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignatureParameter.java

```diff
@@ -83,7 +82,9 @@ public boolean isVariable()
 
     private <A> A getValue(ParameterKind expectedParameterKind, Class<A> target)
     {
-        verify(kind == expectedParameterKind, format("ParameterKind is [%s] but expected [%s]", kind, expectedParameterKind));
```

Comment:
you could use Preconditions.checkState here, it can be something like checkState(kind == expectedParameterKind, "ParameterKind is [%s] but expected [%s]", kind, expectedParameterKind). Then you don't have the cost of String.format(), but it's all in one line.